test/transport/PayloadTest.cpp

```diff
@@ -373,7 +373,8 @@ SCENARIO("test getSerializedSizeOf method") {
     Payload src(Payload::DefaultAllocator(32));
```

Comment:
Maybe just add one more test for bad data. I'd serialize a bad string (e.g. "abcd") and then assert that read should return false when you attempt to read a Decimal50/100